src/components/Table/Table.qa.stories.tsx

```diff
@@ -0,0 +1,332 @@
+
+export default {
+  component: GridTable,
+  title: "Design QA/Table",
+  parameters: { layout: "fullscreen", backgrounds: { default: "white" } },
+} as Meta;
+
+type BeamData = {
+  id: string;
+  favorite: boolean;
+  status: string;
+  commitmentName: string;
+  tradeCategories: string[];
+  location: string;
+  date: string;
+  priceInCents: number;
+};
+type BeamRow = SimpleHeaderAndDataWith<BeamData>;
+
+export function Fixed() {
+  return (
+    <GridTable<BeamRow>
+      style={beamFixedStyle}
+      sorting={{ on: "client", initial: [1, "ASC"] }}
+      columns={beamStyleColumns()}
+      rows={[
+        { kind: "header", id: "header" },
+        ...zeroTo(20).map((idx) => ({
+          kind: "data" as const,
+          id: `r:${idx + 1}`,
+          data: {
+            id: `r:${idx + 1}`,
+            favorite: idx < 5,
+            commitmentName: idx === 9 ? "A longer name that will truncate with an ellipsis" : `Commitment ${idx + 1}`,
+            date: `01/${idx + 1 > 9 ? idx + 1 : `0${idx + 1}`}/2020`,
+            status: "Success",
+            tradeCategories: ["Roofing", "Architecture", "Plumbing"],
+            priceInCents: 1234_56 + idx,
+            location: idx % 2 ? "l:1" : "l:2",
+          },
+        })),
+      ]}
+    />
+  );
+}
+
+export function Flexible() {
+  return (
+    <GridTable<BeamRow>
+      style={beamFlexibleStyle}
+      sorting={{ on: "client", initial: [1, "ASC"] }}
+      columns={beamStyleColumns()}
+      rows={[
+        { kind: "header", id: "header" },
+        ...zeroTo(20).map((idx) => ({
+          kind: "data" as const,
+          id: `r:${idx + 1}`,
+          data: {
+            id: `r:${idx + 1}`,
+            favorite: idx < 5,
+            commitmentName: idx === 9 ? "A longer name that will truncate with an ellipsis" : `Commitment ${idx + 1}`,
+            date: `01/${idx + 1 > 9 ? idx + 1 : `0${idx + 1}`}/2020`,
+            status: "Success",
+            tradeCategories: ["Roofing", "Architecture", "Plumbing"],
+            priceInCents: 1234_56 + idx,
+            location: idx % 2 ? "l:1" : "l:2",
+          },
+        })),
+      ]}
+    />
+  );
+}
+
+type BeamBudgetData = {
+  name?: ReactNode;
+  original?: number;
+  changeOrders?: number;
+  reallocations?: number;
+  revised?: number;
+  committed?: number;
+  difference?: number;
+  actuals?: number;
+  projected?: number;
+  costToComplete?: number;
+  children?: BeamChildRow[];
+};
+type HeaderRow = { kind: "header" };
+type BeamTotalsRow = { kind: "totals"; id: string } & BeamBudgetData;
+type BeamParentRow = { kind: "parent"; id: string } & BeamBudgetData;
+type BeamChildRow = { kind: "child"; id: string } & BeamBudgetData;
+type BeamNestedRow = BeamTotalsRow | HeaderRow | BeamParentRow | BeamChildRow;
+
+export function Nested() {
+  const totalsRowStyles = useMemo(() => ({ totals: { cellCss: beamTotalsRowStyle } }), []);
+  const rowStyles = useMemo(() => ({ parent: { cellCss: beamGroupRowStyle } }), []);
```

Comment:
I'm wondering if there is a way we could avoid a per-file `cellCss` override for these...

I guess `totalsRowStyles` I'm more okay with being one-off...well, or that I think it could eventually move away from a per-file `cellCss` b/c of a pattern like `TableWithTotals rowStyle=fixedStyle totalStyle=totalStyle`, b/c `totalStyle` (with a better name), could define what's currently in `beamTotalsRowStyle` but inside of a dedicated `GridStyle`...

But for the `parent: ...`, I dunno, maybe we could have `GridStyle` know about depth?

Like `GridStyle.rowStyle[0]` is its own `GridStyle` for 1st-level parents, `GridStyle.parentStyle[1]` would be 2nd-level parents....dunno...

Or maybe it's not a whole nested `GridStyle`, but just a few key properties, like:

```
interface GridStyle {
  // existing
  cellCss: ...
  // new
  level: Record<number, { cellCss: Properties, somethingElse: ... }>
}
```

...we could probably also remove `indentOneCss` and `indentTwoCss`, and replace it with this "level 1 has this cellCss, level 2 has that cellCss". Granted, this new approach means a row's level is dictated by whether it's in a `row.children` or not, instead of just deciding to set `indent: 1` out of the blue, but I think that's probably fine. I don't think we ever actually used that flexibility, and it seems like a non-feature.

Reply:
100%. I think there is a ton of clean up we can do with the GridTable styling. Especially now that our rowCss can accomplish a lot more since we removed `display: contents;`. For instance, right now all of our default stylings for rows are applied at each cell. This is fine, but we don't need to do it any longer.
My initial thought/direction was to touch the fewest amount of existing styles as possible, migrate all the existing tables, make the new styles the default styling, and then start trimming off the unused styles. For now I just wanted to get the new styles out there and available.

Reply:
Ah yeah, that's def true for cell css not being as necessary now. Makes sense to minimize churn to get these out.